src/main/java/com/github/kiulian/downloader/model/search/ContinuatedSearchResult.java

```diff
@@ -0,0 +1,21 @@
+package com.github.kiulian.downloader.model.search;
+
+import java.util.List;
+
+public class ContinuatedSearchResult extends SearchResult {
```

Comment:
`ContinuationSearchResult` might be better naming, what do you think?

Reply:
Agree, not pretty, but easy to understand.
I had the same idea, but `ContinuationSearchResult` sounds like a next page result, which is exactly the opposite.
Even though, I would not mind.

Reply:
not clear for me 
>ContinuationSearchResult sounds like a next page result, which is exactly the opposite.
and what actually is it?

Reply:
It's a result that offers a possibility to load a next result page.
At first call, we get an "initial" result, which is not a continuation. It can be "partial" (and carry information to retrieve its continuation) or not.
We can load the next result page, which is a continuation, and can also be "partial" or not (last page).
Maybe `PartialSearchResult` ?
By the way, this class is meant to be "hidden" from user.

Reply:
do we actually need this extra class? I see that you are operating with SearchResult anyway:
```
public RequestSearchContinuation(SearchResult result) {
        super();
        if (!result.hasNext()) {
            throw new IllegalArgumentException("Search result must be continuated");
        }
        this.result = (ContinuatedSearchResult) result;
    }
```    
It can be single class with `continuation()` returning null if `hasNext() == false`

Reply:
Of course, it would be simpler.
The idea was to hide the continuation parameters from the user.
It's strictly technical and does not provide any useful information.

Reply:
@Grodou yes, you are rights, so let's keep it as it is